### ISSUE TITLE
InaSAFE Layer still not correctly uploaded

### ISSUE THREAD

**Issue by lucernae** [kartoza/docker-geosafe#1]
# Scenario

Up the servers, then go to the geonode upload layer page. Upload InaSAFE layers.
In this case, I tried to upload OSM building layers, namely building_osm_4326.shp (and its related files). I got the files from InaSAFE test data.
# Expected

Layer uploaded
# What happened

Layer is correctly processed by QGIS Server (assumed because it created .qgs project files in geonode/qgis_layer directory). But somehow failed when creating thumbnails.
# Stack Trace

You can view geonode's django log by executing `make geosafe-log`. It was attached to process, so you can run that first, then replicate the scenario, and watch the logs.

```
[29/Aug/2016 06:54:14] "GET /layers/upload HTTP/1.0" 200 31181
[29/Aug/2016 06:54:14] "GET /jsi18n/ HTTP/1.0" 200 13253
[29/Aug/2016 06:54:14] "GET /static/geonode/js/upload/FileTypes.js HTTP/1.0" 304 0
[29/Aug/2016 06:54:14] "GET /static/geonode/js/upload/FileType.js HTTP/1.0" 304 0
[29/Aug/2016 06:54:14] "GET /lang.js HTTP/1.0" 200 10016
[29/Aug/2016 06:54:15] "GET /catalogue/opensearch HTTP/1.0" 200 1099
[29/Aug/2016 06:54:25] "GET /upload/progress HTTP/1.0" 404 18318
QGIS Server Pre Save
QGIS Server Post Save
Geonode Layer Path /usr/src/app/geonode/uploaded/layers/buildings_osm_4326_YmNUgl2.shp
Copying /usr/src/app/geonode/uploaded/layers/buildings_osm_4326_YmNUgl2.shp
Success
Copying /usr/src/app/geonode/uploaded/layers/buildings_osm_4326_YmNUgl2.shx
Success
Copying /usr/src/app/geonode/uploaded/layers/buildings_osm_4326_YmNUgl2.dbf
Success
Copying /usr/src/app/geonode/uploaded/layers/buildings_osm_4326_YmNUgl2.prj
Success
Copying /usr/src/app/geonode/uploaded/layers/buildings_osm_4326_YmNUgl2.xml
Success
zip_download_url: /qgis-server/download-zip/buildings_osm_4326
Creating the QGIS Project : http://qgis-server/cgi-bin/qgis_mapserv.fcgi?PROJECT=/usr/src/app/geonode/qgis_layer/buildings_osm_4326_YmNUgl2.qgs&FILES=/usr/src/app/geonode/qgis_layer/buildings_osm_4326_YmNUgl2.shp&NAMES=buildings_osm_4326&SERVICE=MAPCOMPOSITION
tile_url: /qgis-server/tiles/buildings_osm_4326/1234/5678/910.png
tile_url: /qgis-server/tiles/buildings_osm_4326/{z}/{x}/{y}.png
http://localhost/qgis-server/thumbnail/buildings_osm_4326
[Errno 111] Connection refused
Traceback (most recent call last):
  File "/usr/src/app/geonode/layers/views.py", line 171, in layer_upload
    metadata_uploaded_preserve=form.cleaned_data["metadata_uploaded_preserve"]
  File "/usr/src/app/geonode/layers/utils.py", line 494, in file_upload
    defaults=defaults
  File "/usr/local/lib/python2.7/site-packages/django/db/models/manager.py", line 127, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/modeltranslation/manager.py", line 389, in get_or_create
    return super(MultilingualQuerySet, self).get_or_create(**kwargs)
  File "/usr/local/lib/python2.7/site-packages/django/db/models/query.py", line 407, in get_or_create
    return self._create_object_from_params(lookup, params)
  File "/usr/local/lib/python2.7/site-packages/django/db/models/query.py", line 439, in _create_object_from_params
    obj = self.create(**params)
  File "/usr/local/lib/python2.7/site-packages/modeltranslation/manager.py", line 381, in create
    return super(MultilingualQuerySet, self).create(**kwargs)
  File "/usr/local/lib/python2.7/site-packages/django/db/models/query.py", line 348, in create
    obj.save(force_insert=True, using=self.db)
  File "/usr/local/lib/python2.7/site-packages/polymorphic/models.py", line 90, in save
    return super(PolymorphicModel, self).save(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/django/db/models/base.py", line 734, in save
    force_update=force_update, update_fields=update_fields)
  File "/usr/local/lib/python2.7/site-packages/django/db/models/base.py", line 771, in save_base
    update_fields=update_fields, raw=raw, using=using)
  File "/usr/local/lib/python2.7/site-packages/django/dispatch/dispatcher.py", line 189, in send
    response = receiver(signal=self, sender=sender, **named)
  File "/usr/src/geonode_qgis_server/signals.py", line 182, in qgis_server_post_save
    create_thumbnail(instance, thumbnail_remote_url, ogc_client=http_client)
  File "/usr/src/app/geonode/layers/utils.py", line 716, in create_thumbnail
    resp, image = ogc_client.request(thumbnail_create_url)
  File "/usr/local/lib/python2.7/site-packages/httplib2/__init__.py", line 1609, in request
    (response, content) = self._request(conn, authority, uri, request_uri, method, body, headers, redirections, cachekey)
  File "/usr/local/lib/python2.7/site-packages/httplib2/__init__.py", line 1351, in _request
    (response, content) = self._conn_request(conn, request_uri, method, body, headers)
  File "/usr/local/lib/python2.7/site-packages/httplib2/__init__.py", line 1272, in _conn_request
    conn.connect()
  File "/usr/local/lib/python2.7/site-packages/httplib2/__init__.py", line 928, in connect
    raise socket.error, msg
error: [Errno 111] Connection refused
Internal Server Error: /layers/upload
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python2.7/site-packages/django/contrib/auth/decorators.py", line 22, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/usr/src/app/geonode/layers/views.py", line 184, in layer_upload
    upload_session.context = log_snippet(CONTEXT_LOG_FILE)
  File "/usr/src/app/geonode/layers/views.py", line 93, in log_snippet
    if not os.path.isfile(log_file):
  File "/usr/local/lib/python2.7/genericpath.py", line 29, in isfile
    st = os.stat(path)
TypeError: coercing to Unicode: need string or buffer, NoneType found
[29/Aug/2016 06:54:27] "POST /layers/upload HTTP/1.0" 500 49051
[29/Aug/2016 06:58:05] "GET / HTTP/1.0" 200 17862
```

**Comment by lucernae** [comment]
We can look at QGIS Server configuration in:
https://github.com/lucernae/docker-geosafe/blob/develop/src/core/settings.py

**Comment by lucernae** [comment]
Done

**Comment by Gustry** [comment]
@lucernae What do you mean by done ? Did you fix this issue ?
I have the same issue when I want to run geonode master with the qgis backend only.

**Comment by lucernae** [comment]
Reopen this again, it seems somehow the problem comes out again.
The previous fix is done by separating qgis-server service and django, so qgis-server can inherit from ubuntu:xenial image to follow instructions from @Gustry closely. 
However, the problem now is, qgis-server didn't load otf-project plugin at all. I replaced otf-project with HelloServer plugins and it works, but I had to rename the folder as HelloServer. it won't work for any other name, somehow. So, I also tried to rename otf-project to other combination like otfproject, otf_project, OTF-Project (following metadata.txt), OtfProject, and OtfProjectServer. But none works.
